azkustodata/value/value.go

```diff
@@ -81,7 +81,7 @@ func parseError(expected interface{}, actual interface{}, err error) error {
 
 func (p *pointerValue[T]) Unmarshal(i interface{}) error {
 	if i == nil {
-		p.value = nil
+		p.value = (*T)(nil)
```

Comment:
i don't think this is required, since value is typed as *T, can probably just leave as it was